robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowAssetManager.java

```diff
@@ -421,6 +421,86 @@ public void setConfiguration(int mcc, int mnc, String locale,
     return ints;
   }
 
+ protected TypedArray getTypedArrayResource(Resources resources, int resId) {
+    TypedResource value = getAndResolve(resId, RuntimeEnvironment.getQualifiers(), true);
+    if (value == null) {
+      return null;
+    }
+    TypedResource[] items = getConverter(value).getItems(value);
+    return getTypedArray(resources, items, resId);
+  }
+
+  private TypedArray getTypedArray(Resources resources, TypedResource[] typedResources, int resId) {
+    final CharSequence[] stringData = new CharSequence[typedResources.length];
+    final int totalLen = typedResources.length * ShadowAssetManager.STYLE_NUM_ENTRIES;
+    final int[] data = new int[totalLen];
+
+    for (int i = 0; i < typedResources.length; i++) {
+      final int offset = i * ShadowAssetManager.STYLE_NUM_ENTRIES;
+      TypedResource typedResource = typedResources[i];
+
+      // Classify the item.
+      int type = getResType(typedResource);
+      if (type == -1) {
+        // This type is unsupported; leave empty.
+        continue;
+      }
+
+      final TypedValue typedValue = new TypedValue();
+      if (type == TypedValue.TYPE_REFERENCE) {
+        final String resName = typedResource.asString();
+        final int startIdx = resName.indexOf("@");
+        final int slashIdx = resName.indexOf("/");
+        typedValue.resourceId = getResourceIdentifier(resName.substring(slashIdx + 1),
+                resName.substring(startIdx + 1, slashIdx), getResourcePackageName(resId));
+        typedResource = resolve(typedResource, RuntimeEnvironment.getQualifiers(), typedValue.resourceId);
+        // Reclassify to a non-reference type.
+        type = getResType(typedResource);
+        if (type == -1) {
+          // This type is unsupported; leave empty.
+          continue;
+        }
+      }
+
+      getConverter(typedResource).fillTypedValue(typedResource.getData(), typedValue);
+
+      data[offset + ShadowAssetManager.STYLE_TYPE] = type;
```

Comment:
Honestly I don't see a way to do it cleanly, because although the structure is similar, the two methods share very little code.